cmd/reva/app-tokens-create.go

```diff
@@ -0,0 +1,257 @@
+// Copyright 2018-2021 CERN
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// In applying this license, CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+package main
+
+import (
+	"context"
+	"encoding/gob"
+	"io"
+	"reflect"
+	"strings"
+	"time"
+
+	authapp "github.com/cs3org/go-cs3apis/cs3/auth/applications/v1beta1"
+	authpb "github.com/cs3org/go-cs3apis/cs3/auth/provider/v1beta1"
+	gateway "github.com/cs3org/go-cs3apis/cs3/gateway/v1beta1"
+	rpc "github.com/cs3org/go-cs3apis/cs3/rpc/v1beta1"
+	share "github.com/cs3org/go-cs3apis/cs3/sharing/link/v1beta1"
+	provider "github.com/cs3org/go-cs3apis/cs3/storage/provider/v1beta1"
+	types "github.com/cs3org/go-cs3apis/cs3/types/v1beta1"
+	"github.com/cs3org/reva/pkg/auth/scope"
+	"github.com/cs3org/reva/pkg/errtypes"
+)
+
+type appTokenCreateOpts struct {
+	Expiration string
+	Label      string
+	Path       stringSlice
+	Share      stringSlice
+	Unlimited  bool
+}
+
+type stringSlice []string
+
+func (ss *stringSlice) Set(value string) error {
+	*ss = append(*ss, value)
+	return nil
+}
+
+func (ss *stringSlice) String() string {
+	return strings.Join([]string(*ss), ",")
+}
+
+var createOpts *appTokenCreateOpts = &appTokenCreateOpts{}
+
+const layoutTime = "2006-01-02"
+
+func appTokensCreateCommand() *command {
+	cmd := newCommand("token-create")
+	cmd.Description = func() string { return "create a new application tokens" }
+	cmd.Usage = func() string { return "Usage: token-create" }
+
+	cmd.StringVar(&createOpts.Label, "label", "", "set a label")
```

Comment:
Let's use the `cmd.String` method, that's the convention we follow for all the commands. Same for all the flags

Reply:
Ok, it is only to not pass lot of flags to functions that need them, but you can just pussa a struct...

Reply:
We're getting the input from the CLI. How do you input a struct?

Reply:
it is used for example in the getScope function (line 144). rather than pass lot of variable, that could grow in future as soon as new scopes are available, i guess